cas-client-core/src/main/java/org/jasig/cas/client/session/SingleSignOutHandler.java

```diff
@@ -298,18 +295,24 @@ private void destroySession(final HttpServletRequest request) {
 
         final String token = XmlUtils.getTextForElement(logoutMessage, "SessionIndex");
         if (CommonUtils.isNotBlank(token)) {
-            final HttpSession session = this.sessionMappingStorage.removeSessionByMappingId(token);
+            final ClientSession session = this.sessionMappingStorage.removeSessionByMappingId(token);
 
             if (session != null) {
                 final String sessionID = session.getId();
                 logger.debug("Invalidating session [{}] for token [{}]", sessionID, token);
 
                 try {
                     session.invalidate();
-                } catch (final IllegalStateException e) {
+                } catch (final RuntimeException e) {
                     logger.debug("Error invalidating session.", e);
                 }
-                this.logoutStrategy.logout(request);
+
+                try {
+                    request.logout();
```

Comment:
This isn't semantically equivalent to the code you replaced that is sensitive to the servlet environment. In a servlet environment prior to 3.0, this will throw on every logout attempt, which is not what we want. If something happens every time, it's not exceptional. I'm open to the argument that older servlet environments are no longer relevant for future java-cas-client releases, but I did want to note that this is a change in behavior for the noted case.